src/OrchardCore.Modules/OrchardCore.Search/Controllers/SearchController.cs

```diff
@@ -0,0 +1,188 @@
+        var from = 0;
+        var size = pager.PageSize + 1;
+
+        if (pagerParameters.Before != null)
+        {
+            from = Convert.ToInt32(pagerParameters.Before) - pager.PageSize - 1;
+            size = Convert.ToInt32(pagerParameters.Before);
+        }
+        else if (pagerParameters.After != null)
+        {
+            from = Convert.ToInt32(pagerParameters.After);
+            size = Convert.ToInt32(pagerParameters.After) + pager.PageSize + 1;
+        }
+
+        var searchResult = await searchService.SearchAsync(viewModel.Index, viewModel.Terms, from, size);
+
+        if (!searchResult.Success || !searchResult.ContentItemIds.Any())
+        {
+            return View(await GetEmptyShape(viewModel, searchSettings));
+        }
+
+        // Query database to retrieve content items.
+        IQuery<ContentItem> query;
+
+        if (searchResult.Latest)
+        {
+            query = _session.Query<ContentItem, ContentItemIndex>()
+                .Where(x => x.ContentItemId.IsIn(searchResult.ContentItemIds) && x.Latest);
+        }
+        else
+        {
+            query = _session.Query<ContentItem, ContentItemIndex>()
+                .Where(x => x.ContentItemId.IsIn(searchResult.ContentItemIds) && x.Published);
+        }
+
+        // Sort the content items by their position in the search results returned by search service.
+        var containedItems = await query.Take(pager.PageSize + 1).ListAsync();
+
+        // Set the PagerSlim before and after links.
+        if (pagerParameters.After != null || pagerParameters.Before != null)
+        {
+            pager.Before = null;
+
+            if (from + 1 > 1)
+            {
+                pager.Before = (from + 1).ToString();
+            }
+        }
+
+        pager.After = null;
+
+        if (containedItems.Count() == pager.PageSize + 1)
+        {
+            pager.After = (size - 1).ToString();
+        }
+
+        var shape = await _shapeFactory.CreateAsync<SearchIndexViewModel>("Search__List", async model =>
+        {
+            model.PageTitle = searchSettings.PageTitle;
+            model.Terms = viewModel.Terms;
+            model.SearchForm = new SearchFormViewModel("Search__Form")
+            {
+                Terms = viewModel.Terms,
+                Placeholder = searchSettings.Placeholder,
+                Index = viewModel.Index,
+            };
+            model.SearchResults = new SearchResultsViewModel("Search__Results")
+            {
+                ContentItems = containedItems.OrderBy(x => searchResult.ContentItemIds.IndexOf(x.ContentItemId))
+                .Take(pager.PageSize)
+                .ToList(),
+            };
+            model.Pager = (await New.PagerSlim(pager)).UrlParams(new Dictionary<string, string>()
+            {
+                { "Terms", viewModel.Terms },
+                { "Index", viewModel.Index },
+            });
+        });
+
+        return View(shape);
```

Comment:
@MikeAlhayek @sebastienros Just for info we already can override a pure mvc view in a theme without having to create a shape with the `_shapeFactory`, from memory under `Views/ModuleArea/ControllerName/Action.cshtml`.

Or to use a simpler path as for shape, maybe just make `SearchIndexViewModel` inheriting from `ShapeViewModel`, as we already did for `SearchFormViewModel` and `SearchResultsViewModel`.

Reply:
just confirmed that `Views/ModuleArea/ControllerName/Action.cshtml` is the right path too.